freqtrade/rpc/api_server.py

```diff
@@ -0,0 +1,108 @@
+        thread.daemon = True                            # Daemonize thread
+        thread.start()     # Start the execution
+
+
+    def run(self, freqtrade):
+        """ Method that runs forever """
+        self._config = freqtrade.config
+
+        """
+        Define the application routes here
+        each Telegram command should have a like local substitute
+        """
+        @app.route("/")
+        def hello():
+            # For simple rest server testing via browser
+            # cmds = 'Try uri:/daily?timescale=7 /profit /balance /status
+            #         /status /table /performance /count,
+            #         /start /stop /help'
+
+            rest_cmds ='Commands implemented: <br>' \
+                       '<a href=/daily?timescale=7>/daily?timescale=7</a>' \
+                       '<br>' \
+                       '<a href=/stop>/stop</a>' \
+                       '<br>' \
+                       '<a href=/start>/start</a>'
+            return rest_cmds
+
+        @app.route('/daily', methods=['GET'])
+        def daily():
+            try:
+                timescale = request.args.get('timescale')
+                logger.info("LocalRPC - Daily Command Called")
+                timescale = int(timescale)
+
+                stats = self._rpc_daily_profit(timescale,
+                                               self._config['stake_currency'],
+                                               self._config['fiat_display_currency']
+                                               )
+
+                stats = dumps(stats, indent=4, sort_keys=True, default=str)
+                return stats
+            except RPCException as e:
+                return e
+
+        @app.route('/start', methods=['GET'])
+        def start():
+            """
+            Handler for /start.
+            Starts TradeThread
+            """
+            msg = self._rpc_start()
+            print("msg is", msg)
+            return msg
+
+        @app.route('/stop', methods=['GET'])
+        def stop():
+            """
+            Handler for /stop.
+            Stops TradeThread
+            """
+            msg = self._rpc_stop()
+            print("msg is", msg)
+            return msg
+
+        """
+        Section to handle configuration and running of the Rest serve
+        also to check and warn if not bound to 127.0.0.1 as a security risk.
+        """
+        rest_ip = self._config['api_server']['listen_ip_address']
+        rest_port = self._config['api_server']['listen_port']
+
+        if not IPv4Address(rest_ip).is_loopback :
+            logger.info("SECURITY WARNING - Local Rest Server listening to external connections")
+            logger.info("SECURITY WARNING - This is insecure please set to your loopback, e.g 127.0.0.1 in config.json")
+
+        # Run the Server
+        logger.info('Starting Local Rest Server')
+        app.run(host=rest_ip, port=rest_port)
```

Comment:
I don't have experience with flask but I'd maybe make this file return the app so that in the future you might want to run the service separetly with e.g. gunicorn. So something along the lines of
```
app = Flask(__name__)
api = ApiServerSuperWrap(app, freqtrade)
```

Though, I feel that the API and the flask is so tightly coupled that it'll be hard to maintain in the long run. It might be easier to just:
```
app = Flask(__name__)
api = ApiServerSuperWrap(freqtrade)
# add the api routes manually
app.add_url_rule('/', 'hello', api.hello)
```

This way the `app` and the `api` are not coupled and if in the future we might want more powerful http-server (e.g `apistar`) we can easily drop it in. And when you implement it like that you can run the REST endpoint e.g. with `gunicorn`

Reply:
This isn't possible with the current RPCManager, because it only instantiates a RPC class without arguments and I think he shouldn't know more about `ApiServerSuperWrap` than that.

I think with that way wou would end up with the same error because `app.hello` would require `self: RPC` as first argument.